src/controller.cpp

```diff
@@ -914,49 +915,36 @@ int Controller::execute_commands(const std::vector<std::string>& cmds)
 	return EXIT_SUCCESS;
 }
 
-std::string Controller::write_temporary_item(RssItem& item)
+Filepath Controller::write_temporary_item(RssItem& item)
 {
-	char filename[_POSIX_PATH_MAX];
+	Filepath filename_template;
 	char* tmpdir = getenv("TMPDIR");
 	if (tmpdir != nullptr) {
-		snprintf(filename,
-			sizeof(filename),
-			"%s/newsboat-article.XXXXXX",
-			tmpdir);
+		filename_template = Filepath::from_locale_string(tmpdir);
+		filename_template.push("newsboat-article.XXXXXX"_path);
 	} else {
-		snprintf(filename,
-			sizeof(filename),
-			"/tmp/newsboat-article.XXXXXX");
+		filename_template = "/tmp/newsboat-article.XXXXXX"_path;
 	}
-	int fd = mkstemp(filename);
+
+	auto template_string = filename_template.to_locale_string();
+
+	int fd = mkstemp(template_string.data());
 	if (fd != -1) {
+		const auto filename = Filepath::from_locale_string(template_string);
 		write_item(item, filename);
 		close(fd);
-		return std::string(filename);
+		return filename;
 	} else {
-		return "";
+		return {};
 	}
 }
 
-void Controller::write_item(RssItem& item, const std::string& filename)
+void Controller::write_item(RssItem& item, const Filepath& filename)
 {
-	const std::string save_path = cfg.get_configvalue("save-path");
-	auto spath = save_path.back() == '/' ? save_path : save_path + "/";
-
-	std::string path;
-	switch (filename[0]) {
-	case '/':
-		path = filename;
-		break;
-	case '~':
-		path = utils::resolve_tilde(filename);
-		break;
-	default:
-		path = spath + filename;
-		break;
-	}
+	Filepath save_path = cfg.get_configvalue_as_filepath("save-path");
+	save_path.push(utils::resolve_tilde(filename));
```

Comment:
I was wondering if this was correct (given that the old implementation would take `filename` on its own if it was an absolute path)
However, it looks fine now that I checked the Rust documentation on `PathBuf::push()`:
> Extends `self` with `path`.
> If `path` is absolute, it replaces the current path.

Just leaving this comment here for potential other reviewers (or for me in the future)